src/NHibernate/IFutureValueAsync.cs

```diff
@@ -0,0 +1,9 @@
+using System.Threading.Tasks;
+
+namespace NHibernate
+{
+	public interface IFutureValueAsync<T>
+	{
+		Task<T> GetValue();
```

Comment:
>would it be possible to simply add the async GetValue on the currently synchronous IFutureValue?

Why not, [here](https://github.com/OnurGumus/nhibernate-core/commit/327df7a08f559a768840fa7e4554b348941bec0b) you can see the version with `IFutureValue` having also the async `GetValueAsync` method.